pymongo/topology_description.py

```diff
@@ -447,6 +447,15 @@ def _updated_topology_description_srv_polling(topology_description, seedlist):
         return topology_description
 
     # Add SDs corresponding to servers recently added to the SRV record.
+    if topology_description._srv_max_hosts != 0:
```

Comment:
This logic needs to happen *after* the `Remove SDs corresponding to servers no longer part of the SRV record` logic. Otherwise old hosts might never be removed.